library/core/tests/iter.rs

```diff
@@ -3222,3 +3222,85 @@ fn test_flatten_non_fused_inner() {
     assert_eq!(iter.next(), Some(1));
     assert_eq!(iter.next(), None);
 }
+
+#[test]
+fn test_intersperse() {
+    let xs = ["a", "", "b", "c"];
+    let v: Vec<&str> = xs.iter().map(|x| x.clone()).intersperse(", ").collect();
+    let text: String = v.concat();
+    assert_eq!(text, "a, , b, c".to_string());
+
+    let ys = [0, 1, 2, 3];
+    let mut it = ys[..0].iter().map(|x| *x).intersperse(1);
```

Comment:
```suggestion
    let mut it = ys[..0].iter().copied().intersperse(1);
```
> // copied is the same as .map(|&x| x)

Not sure if this works but probably it does.